sds/src/scanner/mod.rs

```diff
@@ -1014,6 +1014,7 @@ mod test {
         assert_eq!(matches.len(), 0);
     }
 
+    #[test]
```

Comment:
Yes, we have warnings generated for this (unused function). We just need to make CI fail on warnings, which also means we need to remove our used of `#[deprecated]` since that generates expected warnings.

Reply:
Could `allow(deprecated)` works for this?

Reply:
https://datadoghq.atlassian.net/browse/SDS-313

Reply:
> Could allow(deprecated) work for this?
Probably. I think it just needs to be in every place it's used, which is probably reasonable for this.